### PR TITLE
add a popup to prevent the user from leaving the page unless the user…

### PR DESCRIPTION
**Pull request by PanthonImem**
… has saved the file within 20 seconds

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 (function () {
+
```
Comment (sorawee) [inline script.js]: Delete this.

File: script.js [hunk 2]
```diff
@@ -43,6 +44,7 @@
 
     let currentId = 1;
 
+
```
Comment (sorawee) [inline script.js]: Ditto.

File: script.js [hunk 3]
```diff
@@ -540,10 +549,14 @@
 
     $('#file-save').click(e => {
         log('Save requested');
+
```
Comment (sorawee) [inline script.js]: Ditto.

File: script.js [hunk 4]
```diff
@@ -540,10 +549,14 @@
...
         const saver = $('#file-save');
         saver.attr('href', 'data:text/csv;charset=utf-8,' + encodeURIComponent(
             Papa.unparse(users.map(user => makeRecord(-1, user, 0, '')).concat(getFreshDatabase()))
         ));
+        window.onbeforeunload = function(e) {
+       };
+       setTimeout(setPopUpTrue, 20000);
```
Comment (sorawee) [inline script.js]: Add a space

File: script.js [hunk 5]
```diff
@@ -540,10 +549,14 @@
 
     $('#file-save').click(e => {
         log('Save requested');
+
         const saver = $('#file-save');
         saver.attr('href', 'data:text/csv;charset=utf-8,' + encodeURIComponent(
             Papa.unparse(users.map(user => makeRecord(-1, user, 0, '')).concat(getFreshDatabase()))
         ));
+        window.onbeforeunload = function(e) {
+       };
```
Comment (sorawee) [inline script.js]: Join this with the previous line.

**Review by sorawee — CHANGES_REQUESTED** [review]

**Comment by PanthonImem** [comment]
solved in the new commit

**Pull request merged**